packages/angular_devkit/build_angular/src/utils/routes-extractor/extractor.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { ApplicationRef, Injector, Type, createPlatformFactory, platformCore } from '@angular/core';
+import {
+  INITIAL_CONFIG,
+  ɵINTERNAL_SERVER_PLATFORM_PROVIDERS as INTERNAL_SERVER_PLATFORM_PROVIDERS,
+} from '@angular/platform-server';
+import { Route, Router, ɵROUTER_PROVIDERS } from '@angular/router';
+import { first } from 'rxjs/operators'; // Import from `/operators` to support rxjs 6 which is still supported by the Framework.
+
+// TODO(alanagius): replace the below once `RouterConfigLoader` is privately exported from `@angular/router`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const RouterConfigLoader = ɵROUTER_PROVIDERS[5] as any;
+type RouterConfigLoader = typeof RouterConfigLoader;
+
+async function* getRoutesFromRouterConfig(
+  routes: Route[],
+  routerConfigLoader: RouterConfigLoader,
+  injector: Injector,
+  parent = '',
+): AsyncIterableIterator<string> {
+  for (const route of routes) {
+    const { path, redirectTo, loadChildren, children } = route;
+    if (path === undefined) {
+      continue;
+    }
+
+    if (redirectTo !== undefined) {
+      // TODO: handle `redirectTo`.
+      continue;
+    }
+
+    if (/[:*]/.test(path)) {
+      // TODO: handle parameterized routes population.
+      continue;
+    }
```

Comment:
I'm wondering if we should collect those routes somewhere in a separate array and output at the end like this (may be if a developer specify a special flag like `--verbose`?):

```
...
Successfully prerendered 1546 routes!

The following routes were skipped because they contain routes with dynamic parameters:

* category/:cid/product/:pid
* product/:id

Learn more about prerendering for such routes at: https://angular.io/guides/prerendering#routes-with-params

The following routes were skipped because they contain redirects:

* path/with/redirect
* another/path/with/redirect

Learn more about prerendering for such routes at: https://angular.io/guides/prerendering#routes-with-redirects
```

Reply:
I like this.